identity-account-storage/src/types/account_id.rs

```diff
@@ -0,0 +1,22 @@
+// Copyright 2020-2022 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+use std::fmt::Display;
+
+use uuid::Uuid;
+
+#[derive(Debug, PartialEq, Eq, Clone, Copy, Hash, serde::Serialize, serde::Deserialize)]
+pub struct AccountId(Uuid);
+
+impl AccountId {
+  /// Creates a new random identifier.
+  pub fn random() -> Self {
+    Self(Uuid::new_v4())
```

Comment:
[`ksuid`](https://docs.rs/ksuid/latest/ksuid/) might be an alternative. It's 20 bytes instead of 16 like `Uuid` or `Ulid`, but also provides that timestamp. I considered it, but couldn't come up with a reason for needing the timestamp. Do we need to offer an API to get an account's creation time? Or do we otherwise need to sort these ids?

Reply:
Not so concerned about the timestamp, more that the ids are sortable/monotonically increasing, since that property may be quite useful for `Storage` implementers (very useful for database performance etc. and could simplify their designs). I actually wondered if the creation timestamp would be a security issue, so it was not the reason for suggesting `ulid`.

I would rather err on the side of `uuid` which is more widely supported and understood rather than use something like `ksuid` just to have the `AccountId` have that property though.